File: Windows/Shiba.Shared/ShibaBinding.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Shiba
{
    internal class ShibaBinding
    {
        public string Path { get; set; }

        public ShibaConverter Converter { get; set; }

        public object Parameter { get; set; }
    }

    internal class ShibaMultiBinding : ShibaBinding
    {
        public string[] Paths { get; set; }
    }
}

File: Android/shiba/src/main/java/moe/tlaster/shiba/ShibaBinding.kt
package moe.tlaster.shiba

import moe.tlaster.shiba.converters.ShibaConverter

open class ShibaBinding(val path: String, val converter: ShibaConverter? = null, val parameter: Any? = null) {
    open fun execute(value: Any?) : Any? {
        return converter?.convert(value, parameter) ?: value
    }
}

class ShibaMultiBinding(val paths: List<String>, converter: ShibaConverter?, parameter: Any?) : ShibaBinding("", converter, parameter)

